frontend/apps/web/components/jobs/NosqlTable/NosqlTable.tsx

```diff
@@ -639,24 +696,61 @@ function IndeterminateCheckbox({
 interface EditDocumentKeyProps {
   text: string;
   onEdit: (updatedObject: { column: string }) => void;
+  data: JobMappingFormValues[];
+  schema: string;
+  table: string;
 }
 
-function EditDocumentKey({ text, onEdit }: EditDocumentKeyProps): ReactElement {
+function EditDocumentKey(props: EditDocumentKeyProps): ReactElement {
+  const { text, onEdit, data, schema, table } = props;
   const [isEditingMapping, setIsEditingMapping] = useState<boolean>(false);
   const [inputValue, setInputValue] = useState<string>(text);
+  const [duplicateError, setDuplicateError] = useState<boolean>(false);
 
   const handleSave = () => {
     onEdit({ column: inputValue });
     setIsEditingMapping(false);
   };
 
+  const handleDocumentKeyChange = (
+    data: JobMappingFormValues[],
+    schema: string,
+    table: string,
+    val: string
+  ) => {
+    const selectedCollection: JobMappingFormValues[] = data.filter(
+      (item: JobMappingFormValues) =>
+        `${item.schema}.${item.table}` === `${schema}.${table}`
+    );
+
+    const isDuplicate = selectedCollection.some(
+      (item: JobMappingFormValues) => item.column === val
+    );
+
+    if (!isDuplicate) {
+      setDuplicateError(false);
+      setInputValue(val);
```

Comment:
looks like this is being called in both branches and should be moved outside of it for clarity.